llvm/lib/Analysis/MemoryDependenceAnalysis.cpp

```diff
@@ -373,7 +373,8 @@ static bool canSkipClobberingStore(const StoreInst *SI,
     return false;
   if (MemoryLocation::get(SI).Size != MemLoc.Size)
     return false;
-  if (std::min(MemLocAlign, SI->getAlign()).value() < MemLoc.Size.getValue())
+  if (std::min(MemLocAlign, SI->getAlign()).value() <
+      MemLoc.Size.getUIntValue())
```

Comment:
I believe this needs to exclude scalable vectors. We don't know whether the alignment is larger than the access size or not.